great_expectations/expectations/metrics/map_metric_provider.py

```diff
@@ -1747,10 +1747,26 @@ def _pandas_map_condition_index(
 
     df = df[boolean_mapped_unexpected_values]
 
-    if result_format["result_format"] == "COMPLETE":
-        return list(df.index)
+    if "unexpected_index_columns" in result_format:
 
-    return list(df.index[: result_format["partial_unexpected_count"]])
+        unexpected_index_list: Optional[List[Dict[str, Any]]] = []
+        unexpected_index_column_names: List[str] = result_format[
+            "unexpected_index_columns"
+        ]
+        unexpected_indices: List[int] = list(df.index)
```

Comment:
This list of unexpected_indices should also respect the partial_unexpected_count directive (i.e. only return the subset)

Reply:
It would also be good to add a test with a partial_unexpected_count set.

Reply:
Good point. Adding now

Reply:
@anthonyburdi I was going through  think the `test_pandas_multiple_unexpected_index_columns_summary_result_format` test does show this. Setting the `result_format` to `SUMMARY` will make only the `partial_unexpected_values` be a part of the output

Reply:
Do you mean `test_pandas_multiple_unexpected_index_column_names_summary_result_format`? I agree, that definitely tests that only partial unexpected counts are returned. I was thinking about two other things: 1. Testing that using a partial_unexpected_count of less than what would normally be returned limits the partial_unexpected_values successfully (if this isn't tested already) 2. More importantly, that the logic here in _pandas_map_condition_index() will always create a list of all the unexpected index values. Would it be possible to only return the partial list so that we aren't storing the full list? Maybe when computing the unexpected_indices list use a limit on the number of values returned.